crates/store/re_log_encoding/src/codec/file/decoder.rs

```diff
@@ -0,0 +1,107 @@
+use super::{MessageHeader, MessageKind};
+use crate::codec::arrow::read_arrow_from_bytes;
+use crate::decoder::DecodeError;
+use crate::{codec::CodecError, Compression};
+use re_log_types::LogMsg;
+use re_protos::TypeConversionError;
+
+impl MessageKind {
+    pub(crate) fn decode(data: &mut impl std::io::Read) -> Result<Self, DecodeError> {
+        let mut buf = [0; 4];
+        data.read_exact(&mut buf)?;
+
+        match u32::from_le_bytes(buf) {
+            1 => Ok(Self::SetStoreInfo),
+            2 => Ok(Self::ArrowMsg),
+            3 => Ok(Self::BlueprintActivationCommand),
+            255 => Ok(Self::End),
+            _ => Err(DecodeError::Codec(CodecError::UnknownMessageHeader)),
+        }
+    }
+}
+
+impl MessageHeader {
+    pub(crate) fn decode(data: &mut impl std::io::Read) -> Result<Self, DecodeError> {
+        let kind = MessageKind::decode(data)?;
+        let mut buf = [0; 4];
+        data.read_exact(&mut buf)?;
+        let len = u32::from_le_bytes(buf);
+
+        Ok(Self { kind, len })
+    }
+}
+
+pub(crate) fn decode(
```

Comment:
can we please add unit tests for all message types, and also for encoding?

Reply:
The encoding/decoding is tested as part of the tests in `re_log_encoding/src/decoder/mod.rs`, but now I realise that isn't thorough enough as it doesn't try to encode an actual chunk with data in it. Will do more testing.

Reply:
thank you!